packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -539,9 +547,14 @@ export class FluidDataStoreRuntime extends EventEmitter implements IFluidDataSto
         );
     }
 
+    // NOTE: Search blob concept!
+    public registerExtraSnapshotContract(contract: ISnapshotContract) {
+        this.extraSnapshotContract = contract;
+    }
+
     public async snapshotInternal(fullTree: boolean = false): Promise<ITreeEntry[]> {
```

Comment:
This is not the right function to instrument - it's not really used (I believe, maybe some legacy flows are using it, like replay tool).
summarize() & getAttachSnapshot() are two APIs called in production